File: Sources/VerseView/VerseView/ViewController.m
//
//  ViewController.m
//  VerseView
//
//  Created by CL Wang on 4/5/23.
//

#import "ViewController.h"
#import "VBVerseViewController.h"

@interface ViewController()

@property VBVerseViewController * verseViewController;
@property NSNumber * BPVNum;

@end

@implementation ViewController
@synthesize BPVNum = BPVNum;

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.BPVNum = @(1001001);
    
    // Do any additional setup after loading the view.
}

- (void)setBPVNum:(NSNumber *)BPVNum {
    self->BPVNum = BPVNum;
    self.verseViewController.verseReference = [[VBVerseReference alloc] initWithVersionMarks:@[] verseIndex:self.BPVNum];
}

- (NSNumber *)BPVNum {
    return self->BPVNum;
}

- (void)prepareForSegue:(NSStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"VBVerseVC"]) {
        self.verseViewController = segue.destinationController;
    }
}

@end

File: Sources/Research Projects/VerseBrowseView/VerseBrowseView/ViewController.swift
//
//  ViewController.swift
//  VerseBrowseView
//
//  Created by CL Wang on 2/6/23.
//

import Cocoa

class ViewController: NSViewController {
    
    var viewBrowseController: VerseBrowseController!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }

    @IBAction func startResizeButtonDidClick(_ sender: NSButton) {
        viewBrowseController.shouldStartResize()
    }
    
    @IBAction func stopResizeButtonDidClick(_ sender: NSButton) {
        viewBrowseController.shouldStopResize()
    }
    
    override func prepare(for segue: NSStoryboardSegue, sender: Any?) {
        if segue.identifier == "VerseBrowseController" {
            viewBrowseController = segue.destinationController as? VerseBrowseController
        }
    }
    
}

